### PR TITLE
Disable some invalid doxygen

### PR DESCRIPTION
**Pull request by marc-hb**
2 commits, main one:

Disable some invalid doxygen comment blocks

Simply replace `/**` with `/*`. This can be very easily reverted if
someone ever fixes the corresponding comment.

We don't build doxygen for .C files in continuous integration, only for
.H files. We don't build doxygen for .C files in continuous integration
because there are way too many errors and no one has time to fix
them. Because we never build doxygen for .C files, people keep adding
untested and invalid doxygen comments or they change the code while
leaving the doxygen out of sync.

I build doxygen for .C files locally. The call graphs are especially
useful (they don't need doxygen comments). I have local hacks to filter
out the most common doxygen errors. The broken doxygen comments disabled
by this commit escaped my local filters.

### CODE REVIEW

File: src/platform/imx8ulp/include/platform/lib/cpu.h [hunk 1]
```diff
@@ -5,7 +5,7 @@
  * Author: Peng Zhang <peng.zhang_8@nxp.com>
  */
 
-/**
+/*
```
Comment (marc-hb) [inline src/platform/imx8ulp/include/platform/lib/cpu.h:8]:
```
sof/src/platform/imx8ulp/include/platform/lib/cpu.h:8: warning: the name 'platform/lib/cpu.h' supplied as the argument in the \file statement matches the following input files:
   sof/src/platform/amd/rembrandt/include/platform/lib/cpu.h
   sof/src/platform/amd/renoir/include/platform/lib/cpu.h
   sof/src/platform/apollolake/include/platform/lib/cpu.h
   sof/src/platform/baytrail/include/platform/lib/cpu.h
   sof/src/platform/cannonlake/include/platform/lib/cpu.h
   sof/src/platform/haswell/include/platform/lib/cpu.h
   sof/src/platform/icelake/include/platform/lib/cpu.h
   sof/src/platform/imx8/include/platform/lib/cpu.h
   sof/src/platform/imx8m/include/platform/lib/cpu.h
   sof/src/platform/imx8ulp/include/platform/lib/cpu.h
   sof/src/platform/library/include/platform/lib/cpu.h
   sof/src/platform/mt8186/include/platform/lib/cpu.h
   sof/src/platform/mt8195/include/platform/lib/cpu.h
   sof/src/platform/suecreek/include/platform/lib/cpu.h
   sof/src/platform/tigerlake/include/platform/lib/cpu.h
Please use a more specific name by including a (larger) part of the path!
```

File: src/audio/module_adapter/module_adapter.c [hunk 2]
```diff
@@ -380,7 +380,7 @@ int module_adapter_params(struct comp_dev *dev, struct sof_ipc_stream_params *pa
 	return 0;
 }
 
-/**
+/*
```
Comment (marc-hb) [inline src/audio/module_adapter/module_adapter.c:383]:
```
sof/src/audio/module_adapter/module_adapter.c:411: warning: Found unknown command '\sink'
sof/src/audio/module_adapter/module_adapter.c:412: warning: Found unknown command '\buff'
sof/src/audio/module_adapter/module_adapter.c:384: warning: Found unknown command '\source'
sof/src/audio/module_adapter/module_adapter.c:385: warning: Found unknown command '\buff'
sof/src/audio/module_adapter/module_adapter.c:386: warning: Found unknown command '\buff_size'
```

File: src/audio/module_adapter/module/iadk_modules.c [hunk 3]
```diff
@@ -202,7 +202,7 @@ static int iadk_modules_free(struct processing_module *mod)
 	return ret;
 }
 
-/**
+/*
```
Comment (marc-hb) [inline src/audio/module_adapter/module/iadk_modules.c:205]:
```
sof/src/audio/module_adapter/module/iadk_modules.c:237: warning: Found unknown command '\fragment'
  parameter 'num_input_buffers'
  parameter 'output_buffers'
  parameter 'num_output_buffers'
sof/src/audio/module_adapter/module/iadk_modules.c:212: warning: Found unknown command '\fragment'
sof/src/audio/module_adapter/module/iadk_modules.c:213: warning: Found unknown command '\params'
sof/src/audio/module_adapter/module/iadk_modules.c:214: warning: Found unknown command '\params'
sof/src/audio/module_adapter/module/iadk_modules.c:214: warning: Found unknown command '\response'
```

File: src/audio/module_adapter/module_adapter.c [hunk 4]
```diff
@@ -407,7 +407,7 @@ ca_copy_from_source_to_module(const struct audio_stream __sparse_cache *source,
 					 MIN(buff_size, tail_size));
 }
 
-/**
+/*
```
Comment (marc-hb) [inline src/audio/module_adapter/module_adapter.c:410]:
```
sof/src/audio/module_adapter/module_adapter.c:384: warning: Found unknown command '\source'
sof/src/audio/module_adapter/module_adapter.c:385: warning: Found unknown command '\buff'
sof/src/audio/module_adapter/module_adapter.c:386: warning: Found unknown command '\buff_size'
sof/src/audio/module_adapter/module_adapter.c:411: warning: Found unknown command '\sink'
sof/src/audio/module_adapter/module_adapter.c:412: warning: Found unknown command '\buff'
sof/src/audio/module_adapter/module_adapter.c:411: warning: Found unknown command '\sink'
sof/src/audio/module_adapter/module_adapter.c:412: warning: Found unknown command '\buff'
sof/src/audio/module_adapter/module_adapter.c:384: warning: Found unknown command '\source'
sof/src/audio/module_adapter/module_adapter.c:385: warning: Found unknown command '\buff'
sof/src/audio/module_adapter/module_adapter.c:386: warning: Found unknown command '\buff_size'
```

File: src/audio/module_adapter/module/generic.c [hunk 5]
```diff
@@ -325,7 +325,7 @@ int module_free(struct processing_module *mod)
 	return ret;
 }
 
-/**
+/*
```
Comment (marc-hb) [inline src/audio/module_adapter/module/generic.c:328]:
```
sof/src/audio/module_adapter/module/generic.c:335: warning: Found unknown command '\fragment'
sof/src/audio/module_adapter/module/generic.c:336: warning: Found unknown command '\params'
sof/src/audio/module_adapter/module/generic.c:337: warning: Found unknown command '\params'
sof/src/audio/module_adapter/module/generic.c:337: warning: Found unknown command '\response'
  parameter 'response_size'
  parameter 'spec'
sof/src/audio/module_adapter/module/generic.c:335: warning: Found unknown command '\fragment'
sof/src/audio/module_adapter/module/generic.c:336: warning: Found unknown command '\params'
sof/src/audio/module_adapter/module/generic.c:337: warning: Found unknown command '\params'
sof/src/audio/module_adapter/module/generic.c:337: warning: Found unknown command '\response'
```

File: src/include/sof/ipc/topology.h [hunk 6]
```diff
@@ -199,12 +199,8 @@ int32_t ipc_comp_pipe_id(const struct ipc_comp_dev *icd);
...
 			void *spec_config);
 
-/**
- * \brief verify component params
- * @param dev Component dev
- * @param flag Component setting flag
- * @param params Ipc params
- * @return verify status
+/*
+ * Warning: duplicate declaration in component.h
```
Comment (lgirdwood) [inline src/include/sof/ipc/topology.h:203]: What happens if we remove the duplicate declaration ?

File: src/audio/kpb.c [hunk 7]
```diff
@@ -157,7 +157,7 @@ static void kpb_lock_init(struct comp_data *kpb)
 
 #endif /* __ZEPHYR__ */
 
-/**
+/*
```
Comment (lgirdwood) [inline src/audio/kpb.c:160]: Why would this need to be reverted if we build doxygen in C files ?

**Review by lgirdwood — COMMENTED** [review]
I agree, having the call graphs by building doxygen from C files adds value, but how much effort is needed to fix ?

File: src/audio/kpb.c (hunk 7 above)
Comment (marc-hb) [reply src/audio/kpb.c:160]: Doxygen looks at `/**` comments only, it ignores single `/*`.

**Comment by marc-hb** [comment]
> I agree, having the call graphs by building doxygen from C files adds value, but how much effort is needed to fix ?

I think most of the  errors with `INPUT = @top_srcdir@/src/` are missing doxygen for some parameters, function prototypes that have changed without the corresponding doxygen update and last but not least "creative" doxygen commands like `\bytes`. It's probably less than a few days work to fix it all HOWEVER it will require experts from various areas; can probably not be done by a single person. Inaccurate documentation is much worse than no documentation.

There are also a number of doxygen comments that should be moved (or merged) from .C files to .H files when the function is not `static`.

But first of all we should wait for our documentation team (@deb-intel, @intelkevinputnam , @anton-intel,...) to help us leave Ubuntu 20.04 behind, see https://github.com/thesofproject/sof-docs/blob/master/.github/workflows/pull-request.yml. No point trying to fix problems with the  obsolete versions of documentation tools currently hardcoded in https://github.com/thesofproject/sof-docs/blob/master/scripts/requirements.txt
I tried a naive bump to Ubuntu 22 once and this caused various failures (forgot which ones).

This PR is just a minimal and easy to revert stop gap.

**Review by RanderWang — APPROVED** [review]
LGTM

**Review by lgirdwood — APPROVED** [review]

**Pull request merged**